mite_cli.py

```diff
@@ -50,40 +51,42 @@ def editor(txt=""):
     editor = os.environ.get("EDITOR", "vi")
     return edit_subprocess(editor, txt)
 
-
-def choose_from_list(lst, key):
-    chosen = ""
-    max_idx = len(lst)+1
-    while not chosen.isnumeric() or not (0 < int(chosen) <= max_idx):
-        chosen = input("> ")
-
-    return lst[int(chosen)-1][key]["id"]
+# choices are of form {(selector) -> id}. Only selector gets displayed to the user.
+# function returns the corresponding ID.
+def choose_with_fzf(choices):
+    chs = iterfzf(choices.keys())
+    return choices[chs]
 
 
 def get_project(mite):
     projects = mite.list_projects()
     customers = mite.list_customers()
-    print(customers)
-    # this is an ugly hack
-    print("Choose a project to add the entry to:")
-    for idx, thng in enumerate(projects):
-        project = thng["project"]
-        customer_name = ""
-        cs = [c for c in customers
-              if c["customer"]["id"] == project["customer_id"]]
-        if cs:
-            customer_name = cs[0]["customer"]["name"]
-        print("[{}] {} ({})".format(idx+1, project["name"], customer_name))
 
-    return choose_from_list(projects, "project")
+    # build {cust id -> name}
+    cust = {}
```

Comment:
Here as well:

```python
cust = {e["id"]:  e["name"] for s in services for e in s["customer"]}
```

Reply:
This one is definitely succinct. I'll change the patch.